hydra-cluster/test/Test/Hydra/Cluster/FaucetSpec.hs

```diff
@@ -12,44 +12,50 @@ import Hydra.Cluster.Faucet (returnFundsToFaucet, seedFromFaucet, seedFromFaucet
 import Hydra.Cluster.Fixture (Actor (..))
 import Hydra.Cluster.Scenarios (EndToEndLog (..))
 import Hydra.Cluster.Util (keysFor)
-import Hydra.Ledger.Cardano (genVerificationKey)
+import Hydra.Ledger.Cardano (genKeyPair, genVerificationKey)
 import Hydra.Logging (showLogsOnFailure)
 import Test.QuickCheck (elements, generate)
 
 spec :: Spec
 spec = do
-  describe "seedFromFaucet" $
-    it "should work concurrently" $
-      showLogsOnFailure "FaucetSpec" $ \tracer ->
-        failAfter 30 $
-          withTempDir "hydra-cluster" $ \tmpDir ->
-            withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node ->
-              replicateConcurrently_ 10 $ do
-                vk <- generate genVerificationKey
-                seedFromFaucet_ node vk 1_000_000 (contramap FromFaucet tracer)
-
-  describe "returnFundsToFaucet" $
-    it "seedFromFaucet and returnFundsToFaucet work together" $ do
-      showLogsOnFailure "FaucetSpec" $ \tracer ->
+  it "seedFromFaucet should work concurrently" $
+    showLogsOnFailure "FaucetSpec" $ \tracer ->
+      failAfter 30 $
+        withTempDir "hydra-cluster" $ \tmpDir ->
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node ->
+            replicateConcurrently_ 10 $ do
+              vk <- generate genVerificationKey
+              seedFromFaucet_ node vk 1_000_000 (contramap FromFaucet tracer)
+  it "seedFromFaucet should work when called multiple times with the same amount of lovelace" $
+    showLogsOnFailure "FaucetSpec" $ \tracer ->
+      failAfter 30 $
         withTempDir "hydra-cluster" $ \tmpDir ->
-          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
-            let faucetTracer = contramap FromFaucet tracer
-            actor <- generate $ elements [Alice, Bob, Carol]
-            (vk, _) <- keysFor actor
-            (faucetVk, _) <- keysFor Faucet
-            initialFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            seeded <- seedFromFaucet node vk 100_000_000 faucetTracer
-            returnFundsToFaucet faucetTracer node actor
-            remaining <- queryUTxOFor networkId nodeSocket QueryTip vk
-            finalFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            foldMap txOutValue remaining `shouldBe` mempty
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node -> do
+            walletVk <- fst <$> generate genKeyPair
+            uTxO1 <- seedFromFaucet node walletVk 2_000_000 (contramap FromFaucet tracer)
+            uTxO2 <- seedFromFaucet node walletVk 2_000_000 (contramap FromFaucet tracer)
+            uTxO1 `shouldNotBe` uTxO2
```

Comment:
Should: merge this with the upper test

Instead of having two (very similar tests as both request the same amount multiple times), we could just improve the upper one with something like:

```haskell
  utxos <- replicateConcurrently 10 $ do
              vk <- generate genVerificationKey
              seedFromFaucet node vk 1_000_000 (contramap FromFaucet tracer)
  -- 10 unique outputs
  length (fold utxos) `shouldBe` 10
```